server/app/mutations/host_nodes/update.rb

```diff
@@ -4,19 +4,81 @@ module HostNodes
   class Update < Mutations::Command
 
     include Common
+    include Logging
 
     required do
       model :host_node
       array :labels, nils: true
     end
 
+    optional do
+      string :availability, in: ['active', 'drain']
+    end
+
     def execute
+      # Update node labels
       self.host_node.labels = self.labels if self.labels
       self.host_node.save
 
       notify_grid(self.host_node.grid)
 
-      self.host_node
+      # Update availability if given and trigger appropriate actions
+      if self.availability && self.host_node.availability != self.availability
+
+        self.host_node.set(:availability => self.availability)
+
+        case self.availability
+        when 'active'
+          start_stateful_services(self.host_node)
```

Comment:
What about this `desired_state: 'stopped' if stateful && host_node.drain?` logic into the `Rpc::ServicePodSerializer`, or perhaps the `GridServiceInstanceDeployer`?

Then this wouldn't need to `stop_stateful_services` / `start_stateful_services`, just trigger deploys for each service on the node. Benefits:

* Errors halfway through these operations here might leave some of the services started/stopped - a deploy would be needed to fix those (re-update would not work)
* This does not actually wait for the services to start/stop, and automation trying to drain a node probably wants to wait for the services to stop - the standard deploy tracking would work for that, if this returned an array of service deploy IDs to track.